src/native/libs/System.Native/pal_networkchange.c

```diff
@@ -38,6 +39,8 @@ Error SystemNative_CreateNetworkChangeListenerSocket(intptr_t* retSocket)
 
 #elif HAVE_RT_MSGHDR
     int32_t sock = socket(PF_ROUTE, SOCK_RAW, 0);
+#else
+    int32_t sock = -1;
```

Comment:
This doesn't do anything, so the last value in `errno` could be success or some failure. We should set it here.

Reply:
Added `EAFNOSUPPORT` `errno`, probably the most relevant when we're dealing with special sockets.